dali/dfuXRefLib/dfuxreflib.cpp

```diff
@@ -917,23 +939,74 @@ static void constructPartname(const char *filename,unsigned n, StringBuffer &pn,
 static bool parseFileName(const char *name,StringBuffer &mname,unsigned &num,unsigned &max,bool &replicate)
 {
     // takes filename and creates mask filename with $P$ extension
-    StringBuffer nonrepdir;
-    replicate = setReplicateDir(name,nonrepdir,false);
-    if (replicate) 
-        name = nonrepdir.str();
+    if (!isContainerized())
+    {
+        // Replicate dir is not supported in containerized environment
+        StringBuffer nonrepdir;
+        replicate = setReplicateDir(name,nonrepdir,false);
+        if (replicate)
+            name = nonrepdir.str();
+    }
+
+    Owned<IPropertyTreeIterator> planesIter = getPlanesIterator("data", nullptr);
+    ForEach(*planesIter)
+    {
+        const IPropertyTree &plane = planesIter->query();
+        const char *prefix = plane.queryProp("@prefix");
+        if (startsWith(name, prefix))
+        {
+            mname.ensureCapacity(strlen(name));
+            mname.append(prefix).append(PATHSEPCHAR);
+            name += strlen(prefix) + 1;
+            if (plane.getPropInt("@numDevices") > 1)
+            {
+                if (*name&&*name!='d')
+                    throw makeStringExceptionV(-1, "numDevices>1 but no stripe sub-directory in file %s", mname.append(name).str());
+                name++;
+                while (*name&&isdigit(*name))
+                    name++;
+                mname.append("d$P$");
```

Comment:
$P$ should be left as meaning the part # only.

Horrible, but if you look at how the result parseFilename 'mname' is used by the caller (i.e. 'orphanname'), it is expecting it to map it against logical file entries (lines 1702-1705), based on CLogicalNameEntry objects created by parseFile (line 1485) whose key value (in manager.logicaldirmap) is 'dirpartmask'. And, 'dirpartmask' is based on the logical files @directory and it's @partmask only.

IOW, it has no concept of stripe or partPerDir at this stage.
It is trying to track physical files found that, in containerized, likely will have stripe and dir per part onto logical file entries that (where stripe nor dirperpart are part specific and the logical file is abstract from that).

So, parseFilename should not be trying to insert stripe or dirperpart into into the returned mname here. It should parse it only.
It may be useful to capture the info. and return the strip # and dir-per-part info as separate info (for future use *maybe*), but adding to the mname will cause every found part here, to fail to match up with the logical file it corresponds to.